kentikapi/internal/api_payloads/saved_filter.go

```diff
@@ -76,7 +77,7 @@ func SavedFilterToPayload(sf models.SavedFilter) savedFilterPayload {
 		FilterLevel:       sf.FilterLevel,
 		CreatedDate:       &sf.CreatedDate,
 		UpdatedDate:       &sf.UpdatedDate,
-		Filters:           FiltersToPayload(sf.Filters),
+		Filters:           filtersToPayload(sf.Filters),
```

Comment:
The logic behind those renamings escapes me. At one place we are changing from (ugly) `camelCase` to (little bit less ugly) `CamelCase` (e.g. [here](https://github.com/kentik/community_sdk_golang/pull/51/files#diff-65234495dfcbcdeb9177db2e28015c5a28515803ef483902a565847f1cbc1ceeR21))... and here it is the other way around. Is this forced by the linters (aren't they fighting each other)? Is this a type-name vs. function-name thing?

Reply:
In Golang the difference between `CamelCase` and `camelCase` is the difference between an `exported` and `unexported` field, struct, function. It is genereally discouraged to have an exported function (a function that can be used by other packages) return an unexported struct. 
Now in this case we had an exported function return an unexported type but the function wasn't used in other packages. So instead of exporting the return I "unexported" (if this is even a word) the function.
In case of RestClient we came to conclusion that because functions returning this struct are used in other packages than the return type should be exported too.
https://github.com/golang/lint/issues/210